x/ecocredit/server/core/query_batches_by_project.go

```diff
@@ -0,0 +1,64 @@
+package core
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types"
+	"github.com/regen-network/regen-ledger/types/ormutil"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+// BatchesByProject queries for all batches in the given credit class.
+func (k Keeper) BatchesByProject(ctx context.Context, request *core.QueryBatchesByProjectRequest) (*core.QueryBatchesByProjectResponse, error) {
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	project, err := k.stateStore.ProjectTable().GetById(ctx, request.ProjectId)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.BatchTable().List(ctx, api.BatchProjectKeyIndexKey{}.WithProjectKey(project.Key), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+
+	batches := make([]*core.BatchInfo, 0)
```

Comment:
```suggestion
	}
	defer it.Close()

	batches := make([]*core.BatchInfo, 0)
```

Reply:
i think this defer line may have been missed in a few areas from previous PR's, we may want to track in an issue or something

Reply:
Updated here and another in `query_batches.go`. https://github.com/regen-network/regen-ledger/pull/1116/commits/168c6130177f6e2bf6b89f704c4e3c16f01aa087